services/src/main/java/io/scalecube/services/ServiceInfo.java

```diff
@@ -0,0 +1,79 @@
+package io.scalecube.services;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Helper class used to register service with tags as metadata in the scalecube cluster.
+ * parsing from service info to json and back.
+ */
+public class ServiceInfo {
+
+  @JsonIgnore
+  private static final ObjectMapper mapper = new ObjectMapper();
```

Comment:
Alternative idea for putting tags into metadata without using JSON is to put key-value pairs in specific format.

E.g.:
You can put following property in metadata. You have one property in metadata which defines service name and the you are looking for specific properties in metadata. For example properties which has keys-values formatted as following:
key format: "<service name>:tag:<tag_key>" 
and value equal to the value of tag with key <tag_key>.

So in such way you can reconstruct out of metadata service instances on other nodes without using JSON. 

And btw having this object mapper here is kind of not nice.